### PR TITLE
Feat(BILL-CPC-247) : Setup Billing Service HTML Elements and UI Front-End Elements

### PR DESCRIPTION
**Pull request by Trim0500**
Would like to merge into main to allow for the billing service to contain new UI elements. We should be able to have a history page documenting all the bills generated and a details page for a given id.

Design Documents:

History Page:

Details Page:

Without css and with mockup data:

image

image

image

Jira:

https://champlainsaintlambert.atlassian.net/browse/CPC-247?atlOrigin=eyJpIjoiMGYyOTM0MWM1MmNjNDI2YmJjZjQ5ZmUyNzkxOWEwODciLCJwIjoiaiJ9

P.S. The reason some people are credited for making changes on the branch is likely due to rebasing from main

### CODE REVIEW

**Comment by Fube** [comment]
Are you planning to do pagination for this?

**Comment by Trim0500** [comment]
I would see the value in providing pagination given a scenario where we get all the bills generated say in an admin page yes. For the time being I wouldn't particularly say it would be of high priority given the small amounts of data being used right now.

File: spring/spring [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
```
Comment (jason-gerard) [inline spring/spring:1]: Do we really want to include this?

File: spring/spring.bat [hunk 2]
```diff
@@ -0,0 +1,76 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (jason-gerard) [inline spring/spring.bat:1]: This one either

**Review by jason-gerard — APPROVED** [review]

File: api-gateway/src/main/resources/static/scripts/bill-details/bill-details.template.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <h3>Final price: {{bills.amount}}</h3>
+    <h3>Visit type: {{bills.visitType}}</h3>
+    <h3>Date generated: {{bills.date}} {{bills.time}}</h3>
+
+    <div>
+        <input type="button" value="Print this page" onclick="printPage()" />
+    </div>
+
+    <button type="submit" formaction="/bills" >Back to History...</button>
+</div>-->
```
Comment (partenersintime) [inline api-gateway/src/main/resources/static/scripts/bill-details/bill-details.template.html:22]: Should we be pushing this commented code to main? Maybe we could implement this in a later story.

File: api-gateway/src/main/resources/static/scripts/bill-details/bill-details.controller.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        var self = this;
+
+        $http.get('api/gateway/bill/' + $stateParams.billId).then(function (resp) {
+            self.bills = resp.data;
+        });
+
+        $http.delete('api/gateway/bill/' + $stateParams.billId).then(function (resp) {
+            self.bills = resp.data;
+        });
+    }]);
```
Comment (partenersintime) [inline api-gateway/src/main/resources/static/scripts/bill-details/bill-details.controller.js:14]: You should maybe consider using a more generalized title to better match what is being added. (Ex: Setup Billing Service Front-End)

File: api-gateway/src/main/resources/static/scripts/bill-history/bill-history.component.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+// Configure a module for the bill history page
+angular.module('billHistory') // Give the module a name
+    .component('billHistory', { // Use the component method to add various scripts and links to a specified module
+        templateUrl: 'scripts/bill/bill-history.template.html', // Add in a page template to append to (bill-history.template.html)
+        controller: 'BillHistoryController' // Add a controller to the module which will tie a HTTP protocol to the page for specific methods
```
Comment (jumpung) [inline api-gateway/src/main/resources/static/scripts/bill-history/bill-history.component.js:7]: Do we need to describe what the code is here?

**Review by LemireCoding — APPROVED** [review]
Improvement Changes were done. Everything looks good

**Review by jumpung — APPROVED** [review]
Successfully added the changes and everything works and looks proper. Good job !!!

**Pull request merged**